src/librustc_mir/transform/const_prop.rs

```diff
@@ -908,17 +954,31 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
             if let Ok(place_layout) = self.tcx.layout_of(self.param_env.and(place_ty)) {
                 let can_const_prop = self.can_const_prop[place.local];
                 if let Some(()) = self.const_prop(rval, place_layout, source_info, place) {
-                    // This will return None if the above `const_prop` invocation only "wrote" a
-                    // type whose creation requires no write. E.g. a generator whose initial state
-                    // consists solely of uninitialized memory (so it doesn't capture any locals).
-                    if let Some(value) = self.get_const(place) {
-                        if self.should_const_prop(value) {
-                            trace!("replacing {:?} with {:?}", rval, value);
-                            self.replace_with_const(rval, value, source_info);
-                            if can_const_prop == ConstPropMode::FullConstProp
-                                || can_const_prop == ConstPropMode::OnlyInsideOwnBlock
-                            {
-                                trace!("propagated into {:?}", place);
+                    if can_const_prop != ConstPropMode::NoPropagation {
+                        // This will return None for variables that are from other blocks,
+                        // so it should be okay to propagate from here on down.
+                        if let Some(value) = self.get_const(place) {
+                            if self.should_const_prop(value) {
+                                trace!("replacing {:?} with {:?}", rval, value);
+                                self.replace_with_const(rval, value, statement.source_info);
+                                if can_const_prop == ConstPropMode::FullConstProp
+                                    || can_const_prop == ConstPropMode::OnlyInsideOwnBlock
+                                {
+                                    trace!("propagated into {:?}", place);
+
+                                    if place.projection.is_empty()
+                                        && !self.has_mutation.contains(place.local)
+                                    {
+                                        self.fully_propagated_locals[place.local] = Some(value);
+                                    }
+                                }
+                            }
+                            if can_const_prop == ConstPropMode::OnlyInsideOwnBlock {
+                                trace!(
+                                    "found local restricted to its block. Will remove it from const-prop after block is finished. Local: {:?}",
+                                    place.local
+                                );
+                                self.locals_of_current_block.insert(place.local);
```

Comment:
Is this some rebase fallout from rebasing over https://github.com/rust-lang/rust/pull/73757 ?

It should not be necessary anymore